### DISCUSSION TITLE
Why static_assert in line 807? I would like to compile with an older version?

### DISCUSSION THREAD

**Discussion by akiy-33** [microsoft/STL#3200] [Q&A] [answered]
https://github.com/microsoft/STL/blob/ea092540b4a3347947863a447d820a9905a68c5b/stl/inc/yvals_core.h#L807

**Comment by MikeGitb** [comment]
Probably, because the STL uses features not available in older versions of clang

**Reply by akiy-33** [reply]
It took me almost a day to downgrade my MSVC Toolset version to compile an older Clang version. Something isn't productive about Microsoft in general. What if vs-installer had a backward compatibility to older LLVM versions. Decided not to think that they would make something like that after forcing programmers to use the newest version of Clang or making them to downgrade their MSVC Toolset version.

**Comment by StephanTLavavej** [comment, marked as answer]
We put a lot of effort into supporting Clang/LLVM as a first-class citizen, but our resources are finite. Supporting older versions of Clang/LLVM would be difficult, as we're continually using the latest compiler features to implement new library features. We also remove compiler bug workarounds, to prevent the library from endlessly accreting complexity. Therefore, each version of the STL supports only the latest version of Clang/LLVM available in the VS installer.
